### PR TITLE
Changed datetime and timestamp types to honor precision and default to 0 digit precision

### PR DESCRIPTION
**Pull request by zachmu**
This matches the MySQL behavior.

Partial fix for https://github.com/dolthub/dolt/issues/6503

### CODE REVIEW

File: enginetest/memory_engine_test.go [hunk 1]
```diff
@@ -275,11 +276,81 @@ func TestSingleQueryPrepared(t *testing.T) {
 
 // Convenience test for debugging a single query. Unskip and set to the desired query.
 func TestSingleScript(t *testing.T) {
-	t.Skip()
-	var scripts = []queries.ScriptTest{}
+	// t.Skip()
```
Comment (Hydrocharged) [inline enginetest/memory_engine_test.go:279]: Probably meant to remove this one since it's in the test file already

File: sql/types/datetime.go [hunk 2]
```diff
@@ -390,9 +404,9 @@ func (t datetimeType) String() string {
 	case sqltypes.Date:
 		return "date"
 	case sqltypes.Datetime:
-		return "datetime(6)"
+		return fmt.Sprintf("datetime(%d)", t.precision)
```
Comment (Hydrocharged) [inline sql/types/datetime.go:407]: If the precision is zero, then we should just display `datetime` and `timestamp`, as `datetime(0)` doesn't look great.

**Review by Hydrocharged — APPROVED** [review]
Overall looks good, just some small comments

File: sql/types/datetime.go (hunk 2 above)
Comment (zachmu) [reply sql/types/datetime.go:407]: Good call

**Pull request merged**